robot/basestation/static/js/elements/navigation/localization.js

```diff
@@ -0,0 +1,193 @@
+/*
+This script uses a lot of p5.js functions and concepts.
+I recommend using the reference: https://p5js.org/reference/#/p5/
+before doing any work on this script as it will greatly facilitate the process.
+*/
+
+$(document).ready(() => {
+    myp5 = new p5(sketch)
+})
+
+let sketch = function(p) {
+    roverHeading = 0
+    roverX = 0 // cartesian coordinates of rover in pixels
+    roverY = 0
+    baseLat = null
+    baseLong = null
+    rover_position_listener.subscribe(function(message) {
+        roverLong = message.x
+        roverLat = message.y
+        roverHeading = message.z
+        initializeBase(message.x, message.y)
+        adjustValuesToCartesian(roverLat, roverLong)
+    })
+
+    dragDiffX = 0
+    dragDiffY = 0
+    zoomValue = 1
+
+    r = 82 // JS keycode
```

Comment:
just call it `r_keycode`. Then you won't need to comment here and where its used to clarify

Reply:
also in drawGrid() you declare an `r` variable so its better to have distinct names imo

Reply:
I was told "nO MaGIc ValuES"

Reply:
Yes that's fine, but the variable name can be confusing. My comment isnt about the values but the name of the variable.

You could write the variable 
 
"R_KEYCODE = 82" (Everywhere where this is use, its clear that this variable is the R keycode)

instead of just writing "r" and then commenting what it does at initatialization. (Some people could see this variable used thinking its for radius or something)

Reply:
I see